### PR TITLE
Added data unset to stop data being read from outdated cookie

### PR DESCRIPTION
**Pull request by gelysis**
Fix for the case when the session increases in size and passes the cookie store limit (of currently 1024 characters).

### CODE REVIEW

File: src/Store/CookieStore.php [hunk 1]
```diff
@@ -137,6 +137,20 @@ public function write($session_id, $session_data)
...
+                Cookie::set(
+                    $this->cookie,
+                    '',
+                    0,
+                    $params['path'],
+                    $params['domain'],
+                    $params['secure'],
+                    $params['httponly']
+                );
+            }
```
[the remark this answers is not in the document]
Comment (gelysis) [reply src/Store/CookieStore.php]: @robbieaverill 
Yes, that clearing the cookie. Makes sense, I will add a comment.

[the remark this answers is not in the document]
Comment (gelysis) [reply src/Store/CookieStore.php]: Done

[the remark this answers is not in the document]
Comment (gelysis) [reply src/Store/CookieStore.php]: @robbieaverill 
I hope the code is now as you expect it ;)

File: src/Store/CookieStore.php [hunk 2]
```diff
@@ -137,6 +137,21 @@ public function write($session_id, $session_data)
             || (strlen($session_data) > static::config()->get('max_length'))
             || !($crypto = $this->getCrypto($session_id))
         ) {
+            if ($this->canWrite() && strlen($session_data) > static::config()->get('max_length')) {
```
Comment (ScopeyNZ) [inline src/Store/CookieStore.php]: Seems to make sense to me. I think it's worth extracting `$this->canWrite()` and `strlen($session_data) > static::config()->get('max_length')` to intermediary variables.

**Comment by gelysis** [comment]
Hi @robbieaverill , hi @ScopeyNZ ,

Do you have a rough idea, when this PR will be merged?

We will need it for a production release early next week. If you need more time to review we will be able to release it from our fork.

Cheers,

**Comment by ScopeyNZ** [comment]
> Do you have a rough idea, when this PR will be merged?

Now 😆 . Although I wish I hadn't because I should've asked you to retarget the `2.1` or `2.0` branches as a patch change. Nevermind I'll cherry-pick it back for you.

Did you want a release to point at too? What version of this module is your project currently looking at?

**Comment by ScopeyNZ** [comment]
Ok the patch exists in `2.0`, `2.1` and `master` now.

For reference:
- `2.0` will be where a future `2.0.1` will be released
- `2.1` for `2.1.1`
- `master` for `2.2.0`

**Comment by gelysis** [comment]
Wow, that's great. Thanks for all, @ScopeyNZ .
Project is pointing at 2.1.x

**Pull request merged**